hack/build/build_image.sh

```diff
@@ -31,13 +31,7 @@ else
   CONTAINER_CMD=docker
 fi
 
-OPERATOR_BUILD_PLATFORM="--platform=linux/amd64"
-if [ -n "${OPERATOR_DEV_BUILD_PLATFORM}" ]; then
-  echo "overriding platform to ${OPERATOR_DEV_BUILD_PLATFORM}"
```

Comment:
just to be super pedantic, IMO better to print that we don't use it anymore and print WARNING if it's set
`echo "WARNING!!! please remove env variable `OPERATOR_DEV_BUILD_PLATFORM `, we don't use it anymore"`

Reply:
the likelihood of someone reading this is minimal